### PR TITLE
Feature/024 bank account list screen

### PR DESCRIPTION
**Pull request by halfangel37**
## Create bank account list screen 👋

- Create bank account list screen 
- Refactor: Fix Profile Behavior, Apply PageContainer to Update Product
- Delete bank account funtionality (its another task but I also do it by mistake)

## Demo 📺

https://www.loom.com/share/f68f6b45da064ff0b05493de53853c75

https://www.loom.com/share/4b26d10f3b374372b2d3c354e68f3292
image
image

### CODE REVIEW

**Comment by luantc13** [comment]
That is weird when we use `,` to format the account number.

image

File: src/companySetting/components/BankAccountsListForm.vue [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      this.dialogTitle = "Confirm delete selected account!";
+      this.dialogContent = "Are you sure you want to delete selected account!";
+      this.isDialog = true;
+    },
+    deleteBankAccount() {
+      this.$emit("deleteBankAccount", {
+        bankAccountId: this.selectedBankAccount.id,
+      });
+    },
+    navigateBankAccount(bankAccount) {
```
Comment (luantc13) [inline src/companySetting/components/BankAccountsListForm.vue:135]: That's not a good name, is `navigateToUpdateBankAccountPage` better?

File: src/companySetting/store/index.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import {
+  getBankAccounts,
+  updateStatusBankAccount,
+  deleteBankAccount,
+} from "@/http/companySetting.js";
+import NProgress from "nprogress";
+import Vue from "vue";
+
+const namespaced = true;
```
Comment (luantc13) [inline src/companySetting/store/index.js:9]: Empty lines to break the code to separate blocks are easier to read in the future.
An example, we can break this file into 6 blocks: `import`, `state`, `mutations`, `actions`, `getters`, and `export`.
Each block is separated by an empty line.

File: src/companySetting/store/index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        NProgress.done();
+        commit("CHANGE_BANK_ACCOUNT_STATUS", {
+          bankAccountId,
+          status: bankAccountStatus,
+        });
+        Vue.$toast.success(statusMessage + " account successfully");
+      })
+      .catch((error) => {
+        NProgress.done();
+        const resetStatus = bankAccountStatus === 1 ? 0 : 1;
```
Comment (luantc13) [inline src/companySetting/store/index.js:67]: We should use `DISABLE_STATUS` and `ENABLE_STATUS` for all places.

**Comment by luantc13** [comment]
For simpler, we can remove three unnecessary tabs.

image

File: src/profile/components/UserProfileForm.vue [hunk 4]
```diff
@@ -56,18 +57,25 @@
...
 import { imageBaseUrl } from "@/config.js";
+import _ from "lodash";
 export default {
   props: {
     user: Object,
     isButtonDisabled: Boolean,
   },
   data() {
     return {
+      profileCopy: "",
```
Comment (luantc13) [inline src/profile/components/UserProfileForm.vue:68]: Why do you want to use an empty string `""` as a default value? `profileCopy` is an object, so the default value should be `undefined`.

File: src/http/companySetting.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { httpClient } from "@/http/httpClient";
```
Comment (luantc13) [inline src/http/companySetting.js:1]: This http service should be `bankAccount.js`.

File: src/companySetting/components/BankAccountsListForm.vue [hunk 6]
```diff
@@ -0,0 +1,149 @@
+<template>
```
Comment (luantc13) [inline src/companySetting/components/BankAccountsListForm.vue:1]: `companySettings` is a better choice, it should be apply for all.

File: src/router/index.js [hunk 7]
```diff
@@ -96,7 +96,30 @@ const routes = [
...
+                        })
+                        .then(() => {
+                          next();
+                        })
+                        .catch((error) => {
+                          console.log(error);
+                        });
+                    },
+                  },
+                ],
```
Comment (luantc13) [inline src/router/index.js:120]: Why do we need `children` with only one child path?

Could you try replacing `import("../views/Setting.vue")` with `import("../companySetting/views/CompanySetting.vue")`?

**Review by luantc13 — CHANGES_REQUESTED** [review]
Great work!
Having some comments, happy to merge when they are resolved.

**Comment by truonghungit** [comment]
Missing the page title 
image

File: src/router/index.js (hunk 7 above)
Comment (truonghungit) [reply src/router/index.js:120]: We should remove `src/views`, please move it into company setting module

File: src/views/Setting.vue [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div>
-    <h1>Setting</h1>
+    <router-view></router-view>
   </div>
 </template>
```
Comment (truonghungit) [inline src/views/Setting.vue:6]: please delete this file if we don't use

File: src/views/Dashboard.vue [hunk 9]
```diff
@@ -16,8 +16,5 @@ export default {
     NavBar,
     MainContent,
   },
-  created() {
-    this.$store.dispatch("profile/getUserProfile");
-  },
 };
 </script>
```
Comment (truonghungit) [inline src/views/Dashboard.vue:20]: remove this file

**Review by truonghungit — CHANGES_REQUESTED** [review]

**Comment by truonghungit** [comment]
image

File: src/companySettings/views/CompanySettings.vue [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+        slider-color="#E7AC28"
+        slider-size="3"
+      >
+        <v-tab v-for="item in items" :key="item">
+          {{ item }}
+        </v-tab>
+      </v-tabs>
+      <v-tabs-items v-model="tab">
+        <v-tab-item v-for="item in items" :key="item">
+          <div class="container" :class="{ active: tab === 1 }">
```
[the remark this answers is not in the document]
Comment (halfangel37) [reply src/companySettings/views/CompanySettings.vue:18]: I changed it to this , it for active the content when click on the tab bank account.
image

**Review by luantc13 — APPROVED** [review]
Great work!

**Review by truonghungit — APPROVED** [review]

**Pull request merged**